aiven/client/client.py

```diff
@@ -400,6 +400,47 @@ def get_service_topic(self, project, service, topic):
         path = self.build_path("project", project, "service", service, "topic", topic)
         return self.verify(self.get, path, result_key="topic")
 
+    def list_m3_namespaces(self, project, service):
+        service_res = self.get_service(project=project, service=service)
+        return service_res.get("user_config", {}).get("namespaces", [])
+
+    def delete_m3_namespace(self, project, service, ns_name):
+        service_res = self.get_service(project=project, service=service)
+        old_namespaces = service_res.get("user_config", {}).get("namespaces", [])
+        new_namespaces = [ns for ns in old_namespaces if ns["name"] != ns_name]
+        if len(old_namespaces) == len(new_namespaces):
+            raise KeyError(f"Namespace '{ns_name}' does not exist")
+        return self.verify(
+            self.put,
+            self.build_path("project", project, "service", service),
+            body={"user_config": {
+                "namespaces": new_namespaces
+            }},
+        )
+
+    def add_m3_namespace(self, project, service, ns_name, ns_type, ns_ret, ns_res=None):
```

Comment:
Maybe we could check some of the values for validity here. For example, the namespace type (aggregated/unaggregated) seems to be possible to accidentally type wrong and it is stored to the user config as the API does not seem to validate it yet at least.

Reply:
Could add here. I think it would be more critical for the API to validate incoming stuff though.

Reply:
added validations